File: leetcode/js/minimum-time-visiting-all-points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function (points) {
  let result = 0;

  for (let i = 0; i < points.length - 1; i++) {
    let [curX, curY] = points[i];
    const [nextX, nextY] = points[i + 1];

    result += Math.max(Math.abs(nextX - curX), Math.abs(nextY - curY));
  }

  return result;
};

File: leetcode/python/minimum-time-visiting-all-points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: list[list[int]]) -> int:
        result = 0

        for i in range(len(points) - 1):
            cur_x, cur_y = points[i]
            next_x, next_y = points[i + 1]

            result += max(abs(next_x - cur_x), abs(next_y - cur_y))

        return result